spec/guest-window-manager-spec.ts

```diff
@@ -2,195 +2,330 @@ import { BrowserWindow } from 'electron';
+    });
+
+    afterEach(closeAllWindows);
 
-  it('does not hang parent window when denying window.open', async () => {
-    browserWindow.webContents.setWindowOpenHandler(() => ({ action: 'deny' }));
-    browserWindow.webContents.executeJavaScript("window.open('https://127.0.0.1')");
-    expect(await browserWindow.webContents.executeJavaScript('42')).to.equal(42);
+    it('spawns browser window when createWindow is provided', async () => {
+      const browserWindowTitle = 'Child browser window';
+
+      const childWindow = await new Promise<Electron.BrowserWindow>(resolve => {
+        browserWindow.webContents.setWindowOpenHandler(() => {
+          return {
+            action: 'allow',
+            createWindow: () => {
+              const childWindow = new BrowserWindow({ title: browserWindowTitle });
+              resolve(childWindow);
+              return childWindow.webContents;
+            }
+          };
+        });
+
+        browserWindow.webContents.executeJavaScript("window.open('about:blank', '', 'show=no') && true");
+      });
+
+      expect(childWindow.title).to.equal(browserWindowTitle);
+    });
+
+    it('spawns browser window with overriden options', async () => {
+      const childWindow = await new Promise<Electron.BrowserWindow>(resolve => {
+        browserWindow.webContents.setWindowOpenHandler(() => {
+          return {
+            action: 'allow',
+            overrideBrowserWindowOptions: {
+              width: 640,
+              height: 480
+            },
+            createWindow: (options) => {
+              expect(options.width).to.equal(640);
+              expect(options.height).to.equal(480);
+              const childWindow = new BrowserWindow(options);
+              resolve(childWindow);
+              return childWindow.webContents;
+            }
+          };
+        });
+
+        browserWindow.webContents.executeJavaScript("window.open('about:blank', '', 'show=no') && true");
+      });
+
+      const size = childWindow.getSize();
+      expect(size[0]).to.equal(640);
+      expect(size[1]).to.equal(480);
+    });
+
+    it('spawns browser window with access to opener property', async () => {
+      const childWindow = await new Promise<Electron.BrowserWindow>(resolve => {
+        browserWindow.webContents.setWindowOpenHandler(() => {
+          return {
+            action: 'allow',
+            createWindow: (options) => {
+              const childWindow = new BrowserWindow(options);
+              resolve(childWindow);
+              return childWindow.webContents;
+            }
+          };
+        });
+
+        browserWindow.webContents.executeJavaScript(`window.open('${url}/child', '', 'show=no') && true`);
+      });
+
+      await once(childWindow.webContents, 'ready-to-show');
+      const childWindowOpenerTitle = await childWindow.webContents.executeJavaScript('window.opener.document.title');
+      expect(browserWindow.title).to.equal(childWindowOpenerTitle);
```

Comment:
value under test goes first, expected value goes last
```suggestion
      expect(childWindowOpenerTitle).to.equal(browserWindow.title);
```